pkg/prometheus/promcfg.go

```diff
@@ -149,7 +149,7 @@ func generateConfig(p *v1.Prometheus, mons map[string]*v1.ServiceMonitor, basicA
 	var additionalScrapeConfigsYaml []yaml.MapSlice
 	err = yaml.Unmarshal([]byte(additionalScrapeConfigs), &additionalScrapeConfigsYaml)
 	if err != nil {
-		errors.Wrap(err, "unmarshalling additional scrape configs failed")
+		return errors.Wrap(err, "unmarshalling additional scrape configs failed")
```

Comment:
I forgot to check the function signature. You would need to do:

```go
return nil, errors.Wrap(err, "unmarshalling additional scrape configs failed")
```

(See [above](https://github.com/coreos/prometheus-operator/pull/1508/files#diff-d7fb441eb132cd7782e17b92b0eb5bcdR97))